linalg/operator.cpp

```diff
@@ -464,11 +466,30 @@ void ConstrainedOperator::Mult(const Vector &x, Vector &y) const
    auto d_x = x.Read();
    // Use read+write access - we are modifying sub-vector of y
    auto d_y = y.ReadWrite();
-   MFEM_FORALL(i, csz,
+   switch (diag_policy)
    {
-      const int id = idx[i];
-      d_y[id] = d_x[id];
-   });
+      case DIAG_ONE:
+         MFEM_FORALL(i, csz,
+         {
+            const int id = idx[i];
```

Comment:
Is there a reason to define `id`, or should these 2 lines be shortened to 1? Also in the `DIAG_ZERO` case.

Reply:
Not a particular reason, as far as I can tell. I kept the style there as it was previously. Somebody may have written it this way for better readability. If you prefer, I can shorten this one as well as the other instances (a total of five) to one line.

Reply:
I don't have a strong preference. For more complicated cases, it makes sense to define a variable like this, but in this case I would slightly prefer shortening the code. Either way is fine with me.